### PR TITLE
Ak/prepare for 0.2.0

### PR DESCRIPTION
**Pull request by akosyakov**

### CODE REVIEW

**Comment by akosyakov** [comment]
Please review.

File: org.eclipse.lsp4j.jsonrpc/src/main/java/org/eclipse/lsp4j/jsonrpc/json/adapters/MessageTypeAdapterFactory.java [hunk 1]
```diff
@@ -88,16 +88,20 @@ public Message read(JsonReader in) throws IOException {
...
+					if (parameterType == null || parameterType == Void.class) {
 						params = new JsonParser().parse(in);
-					else
-						params = gson.fromJson(in, type);
+					} else {
+						params = gson.fromJson(in, parameterType);
+						if (params instanceof JsonElement) {
+							params = gson.fromJson((JsonElement) params, parameterType);
+						}
+					}
```
Comment (spoenemann) [inline org.eclipse.lsp4j.jsonrpc/src/main/java/org/eclipse/lsp4j/jsonrpc/json/adapters/MessageTypeAdapterFactory.java]: The reason I had deferred the parsing of JsonElement to `createMessage(..)` is that I'd like to support an arbitrary order of message attributes. This is relevant when `params` appears before `method`. I vote for restoring the previous behavior, except for the special handling of `Void.class`.

Comment (akosyakov) [inline org.eclipse.lsp4j.jsonrpc/src/main/java/org/eclipse/lsp4j/jsonrpc/json/adapters/MessageTypeAdapterFactory.java]: ok, will do

**Comment by akosyakov** [comment]
@spoenemann I've put deferred params parsing back and added tests for it

**Pull request merged**